odoo/models.py

```diff
@@ -2115,7 +2115,11 @@ def _read_group_process_groupby(self, gb, query):
             }
             if tz_convert:
                 qualified_field = "timezone('%s', timezone('UTC',%s))" % (self._context.get('tz', 'UTC'), qualified_field)
-            qualified_field = "date_trunc('%s', %s::timestamp)" % (gb_function or 'month', qualified_field)
+            days_offset = 0
+            if gb_function == 'week':
+                locale = babel.Locale(get_lang(self.env).code)
+                days_offset = locale.first_week_day and 7 - locale.first_week_day
+            qualified_field = "date_trunc('%s', %s::timestamp) + INTERVAL '-%s DAY'" % (gb_function or 'month', qualified_field, days_offset)
```

Comment:
BTW, I would "inject" the value of `day_offset` as `%d` (to avoid any accidental SQL injection).